util/predicates/cluster_predicates.go

```diff
@@ -188,3 +188,27 @@ func ClusterUnpausedAndInfrastructureReady(logger logr.Logger) predicate.Funcs {
 	// Use any to ensure we process either create or update events we care about
 	return Any(log, createPredicates, updatePredicates)
 }
+
+// ClusterIsManagedTopology returns a Predicate that returns true when cluster.Spec.Topology
+// is NOT nil and false otherwise.
+func ClusterIsManagedTopology(logger logr.Logger) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			log := logger.WithValues("predicate", "ClusterIsManagedTopology", "eventType", "update")
+			cluster := &clusterv1.Cluster{}
+
+			log.WithValues("namespace", cluster.Namespace, "cluster", cluster.Name)
+
+			if cluster.Spec.Topology != nil {
+				log.V(6).Info("Cluster has topology, allowing reconcile")
```

Comment:
Do we need this log line?

Reply:
If not, i'll remove this. Should I go for it?

Reply:
This follows the pattern we have in other predicates, so I think we should keep it, except we want to drop all of them (preferably in a separate PR). But I think it's good to have them and log level 6 is pretty much tracing level so it should be fine.